src/openforms/js/components/form/fieldset.js

```diff
@@ -0,0 +1,33 @@
+import {Formio} from 'formiojs';
+import {FIELDSET_BASIC} from './edit/tabs';
+
+const FormioFieldSet = Formio.Components.components.fieldset;
+
+class FieldSet extends FormioFieldSet {
+    constructor(component, options, data) {
+        /* Field sets have both a 'legend' and a 'label' field. The label should be automatically populated to be equal
+        to the legend, but this doesn't seem to work in Formio 4.13.x. Since in open-forms we always use the label, in
+        this custom fieldset the legend is hidden and automatically filled with the value of the label.
+         */
+        if (component.label) {
+            component.legend = component.label;
+        }
```

Comment:
Does this code run when you're changing the label? It seems odd to put this in the constructor

Reply:
Yes, when the label property is changed in the edit form, the constructor is called and the legend gets updated. I agree though that it feels like a hack